modules/tinymce/src/core/main/ts/fmt/ListItemFormat.ts

```diff
@@ -32,25 +32,44 @@ const isRngEndAtEndOfElement = (rng: Range, elm: Element) => {
 
 const isEditableListItem = (dom: DOMUtils) => (elm: Element) => NodeType.isListItem(elm) && dom.isEditable(elm);
 
+const findParentListItem = (selection: EditorSelection, el: Element) => Optional.from(selection.dom.getParent(el, NodeType.isListItem));
+
 const getFullySelectedBlocks = (selection: EditorSelection) => {
   const blocks = selection.getSelectedBlocks();
   const rng = selection.getRng();
 
   if (selection.isCollapsed()) {
     return [];
-  } if (blocks.length === 1) {
+  }
+
+  // TINY-13197: The first block element returned by getSelectedBlocks() is not LI if the content is wrapped inside a block element, even when the content is fully selected. The following logic finds and replaces the first block with its parent LI
+  if (blocks.length > 0 && isRngStartAtStartOfElement(rng, blocks[0]) && !NodeType.isListItem(blocks[0])) {
+    findParentListItem(selection, blocks[0]).each((listItem) => {
+      blocks[0] = listItem;
+    });
+  }
+
+  if (blocks.length === 1) {
     return isRngStartAtStartOfElement(rng, blocks[0]) && isRngEndAtEndOfElement(rng, blocks[0]) ? blocks : [];
   } else {
     const first = Arr.head(blocks).filter((elm) => isRngStartAtStartOfElement(rng, elm)).toArray();
     const last = Arr.last(blocks).filter((elm) => isRngEndAtEndOfElement(rng, elm)).toArray();
     const middle = blocks.slice(1, -1);
-
     return first.concat(middle).concat(last);
   }
 };
 
 export const getFullySelectedListItems = (selection: EditorSelection): Element[] =>
   Arr.filter(getFullySelectedBlocks(selection), isEditableListItem(selection.dom));
 
-export const getPartiallySelectedListItems = (selection: EditorSelection): Element[] =>
-  Arr.filter(selection.getSelectedBlocks(), isEditableListItem(selection.dom));
+export const getPartiallySelectedListItems = (selection: EditorSelection): Element[] => {
+  const blocks = selection.getSelectedBlocks();
+  // TINY-13197 The first block element returned by getSelectedBlocks() is not LI if the content is wrapped inside a block element, even when the content is fully selected. The following logic finds and replaces the first block with its parent LI
+  if (blocks.length > 0 && !NodeType.isListItem(blocks[0])) {
+    findParentListItem(selection, blocks[0]).each((listItem) => {
+      blocks[0] = listItem;
+    });
+  }
```

Comment:
de-dupe with the fully selected code please. Looks like the only difference is the start of element check.